projects/entity-data-api/src/Mirza/EntityDataAPI/Main.hs

```diff
@@ -0,0 +1,56 @@
+module Mirza.EntityDataAPI.Main (main) where
+
+import           Options.Applicative
+
+import           Network.HTTP.Client           (defaultManagerSettings,
+                                                newManager)
+
+import           Mirza.EntityDataAPI.AuthProxy (runAuthProxy)
+import           Mirza.EntityDataAPI.Types
+
+import           Network.HTTP.ReverseProxy     (ProxyDest (..))
+import qualified Network.Wai.Handler.Warp      as Warp
+
+import qualified Data.ByteString.Char8         as B
+
+import           Crypto.JOSE.JWK               (KeyMaterialGenParam (..),
+                                                genJWK)
+
+data Opts = Opts
+  { _myServiceInfo   :: ServiceInfo
+  , _destServiceInfo :: ServiceInfo
+  , _keySize         :: Int
+  }
+
+main :: IO ()
+main = launchProxy =<< execParser opts where
+  opts = info (optsParser <**> helper)
+    (fullDesc
+    <> progDesc "Reverse proxy for Mirza services"
+    <> header "Entity Data API")
+
+initContext :: Opts -> IO AuthContext
+initContext (Opts myService (destHost, destPort) kSize) = do
+  mngr <- newManager defaultManagerSettings
```

Comment:
Will likely need to be a TLS client manager, http://hackage.haskell.org/package/http-client-tls/docs/Network-HTTP-Client-TLS.html